Library/Homebrew/cmd/uninstall.rb

```diff
@@ -19,12 +19,16 @@ module Homebrew
   def uninstall_args
     Homebrew::CLI::Parser.new do
       usage_banner <<~EOS
-        `uninstall`, `rm`, `remove` [<options>] <formula>
+        `uninstall`, `rm`, `remove` [<options>] <formula>|<cask>
 
-        Uninstall <formula>.
+        Uninstall a <formula> or <cask>.
       EOS
       switch "-f", "--force",
-             description: "Delete all installed versions of <formula>."
+             description: "Delete all installed versions of <formula>. Uninstall even if <cask> is not " \
+                          "installed, overwrite existing files and ignore errors when removing files."
+      switch "--zap",
+             description: "Remove all files associated with the given <formula> or <cask>. " \
```

Comment:
```suggestion
             description: "Remove all files associated with the given <cask>. " \
```

and conflict with `--formula` until this is supported?

Reply:
Well, `--zap` for casks also implies `uninstall`, so I think the same should be the case for formulae, even if no other files are removed yet for formulae.

Reply:
I think people will get confused if it does nothing for formulae, though.

Reply:
Couldn’t we just add a `zap isn’t yet supported on formulae` message when applicable?

Reply:
Maybe a warning that it's a no-op for now.

Reply:
I think it would be good to have a hard failure until it does anything. We have no plans to make `zap` work for formulae and it'll be a non-trivial amount of work to get there (and changing this flag back to reference formulae would be the easiest part of it).